### PR TITLE
Allows configclass `to_dict` operation to handle a list of configclasses

### LINKED ISSUE #1219: [Bug Report] Stateful observation modifiers cannot be added to the observation term config
### Describe the bug

Stateful observation modifiers cannot be added to the observation term config

### Steps to reproduce
Add a `DigitalFilterCfg` to the modifiers list for one of the observation terms:
```omni/isaac/lab_tasks/manager_based/locomotion/velocity/velocity_env_cfg.py```:
```
joint_pos = ObsTerm(
    func=mdp.joint_pos_rel, 
    modifiers=[DigitalFilterCfg(A=[0.0], B=[0.0, 1.0])],
    noise=Unoise(n_min=-0.01, n_max=0.01))
```
Then run training:
```
./isaaclab.sh -p source/standalone/workflows/rsl_rl/train.py --task Isaac-Velocity-Flat-G1-v0
```
This results in the following error:
```
[8.363s] Simulation App Startup Complete
[INFO]: Parsing configuration from: <class 'omni.isaac.lab_tasks.manager_based.locomotion.velocity.config.g1.flat_env_cfg.G1FlatEnvCfg'>
[INFO]: Parsing configuration from: omni.isaac.lab_tasks.manager_based.locomotion.velocity.config.g1.agents.rsl_rl_ppo_cfg:G1FlatPPORunnerCfg
Traceback (most recent call last):
  File "/home/swelch/isaaclab/source/standalone/workflows/rsl_rl/train.py", line 148, in <module>
    main()
  File "/home/swelch/isaaclab/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/hydra.py", line 75, in wrapper
    env_cfg, agent_cfg = register_task_to_hydra(task_name, agent_cfg_entry_point)
  File "/home/swelch/isaaclab/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/utils/hydra.py", line 53, in register_task_to_hydra
    ConfigStore.instance().store(name=task_name, node=cfg_dict)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/hydra/core/config_store.py", line 85, in store
    cfg = OmegaConf.structured(node)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 125, in structured
    return OmegaConf.create(obj, parent, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 178, in create
    return OmegaConf._create_impl(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 900, in _create_impl
    format_and_raise(node=None, key=None, value=None, msg=str(e), cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 861, in _create_impl
    return DictConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 109, in __init__
    self._set_value(content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 690, in _set_value_impl
    self.__setitem__(k, v)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 314, in __setitem__
    self._format_and_raise(key=key, value=value, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 308, in __setitem__
    self.__set_impl(key=key, value=value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 318, in __set_impl
    self._set_item_impl(key, value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 620, in _set_item_impl
    self._wrap_value_and_set(key, value, target_type_hint)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 636, in _wrap_value_and_set
    self._format_and_raise(key=key, value=val, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 628, in _wrap_value_and_set
    wrapped = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1004, in _node_wrap
    node = DictConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 109, in __init__
    self._set_value(content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 690, in _set_value_impl
    self.__setitem__(k, v)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 314, in __setitem__
    self._format_and_raise(key=key, value=value, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 308, in __setitem__
    self.__set_impl(key=key, value=value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 318, in __set_impl
    self._set_item_impl(key, value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 620, in _set_item_impl
    self._wrap_value_and_set(key, value, target_type_hint)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 636, in _wrap_value_and_set
    self._format_and_raise(key=key, value=val, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 628, in _wrap_value_and_set
    wrapped = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1004, in _node_wrap
    node = DictConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 109, in __init__
    self._set_value(content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 690, in _set_value_impl
    self.__setitem__(k, v)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 314, in __setitem__
    self._format_and_raise(key=key, value=value, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 308, in __setitem__
    self.__set_impl(key=key, value=value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 318, in __set_impl
    self._set_item_impl(key, value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 620, in _set_item_impl
    self._wrap_value_and_set(key, value, target_type_hint)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 636, in _wrap_value_and_set
    self._format_and_raise(key=key, value=val, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 628, in _wrap_value_and_set
    wrapped = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1004, in _node_wrap
    node = DictConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 109, in __init__
    self._set_value(content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 690, in _set_value_impl
    self.__setitem__(k, v)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 314, in __setitem__
    self._format_and_raise(key=key, value=value, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 308, in __setitem__
    self.__set_impl(key=key, value=value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 318, in __set_impl
    self._set_item_impl(key, value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 620, in _set_item_impl
    self._wrap_value_and_set(key, value, target_type_hint)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 636, in _wrap_value_and_set
    self._format_and_raise(key=key, value=val, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 628, in _wrap_value_and_set
    wrapped = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1004, in _node_wrap
    node = DictConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 109, in __init__
    self._set_value(content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 690, in _set_value_impl
    self.__setitem__(k, v)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 314, in __setitem__
    self._format_and_raise(key=key, value=value, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 308, in __setitem__
    self.__set_impl(key=key, value=value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 318, in __set_impl
    self._set_item_impl(key, value)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 620, in _set_item_impl
    self._wrap_value_and_set(key, value, target_type_hint)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 636, in _wrap_value_and_set
    self._format_and_raise(key=key, value=val, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 628, in _wrap_value_and_set
    wrapped = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1017, in _node_wrap
    node = ListConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 82, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 80, in __init__
    self._set_value(value=content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 618, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 614, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 659, in _set_value_impl
    self.append(item)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 301, in append
    self._format_and_raise(key=index, value=item, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/listconfig.py", line 298, in append
    self._set_item_impl(index, item)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 620, in _set_item_impl
    self._wrap_value_and_set(key, value, target_type_hint)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 636, in _wrap_value_and_set
    self._format_and_raise(key=key, value=val, cause=e)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/base.py", line 231, in _format_and_raise
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/basecontainer.py", line 628, in _wrap_value_and_set
    wrapped = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1027, in _node_wrap
    node = DictConfig(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 94, in __init__
    self._set_value(content, flags=flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 674, in _set_value_impl
    data = get_structured_config_data(value, allow_objects=ao)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 495, in get_structured_config_data
    return get_dataclass_data(obj, allow_objects=allow_objects)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 421, in get_dataclass_data
    format_and_raise(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 899, in format_and_raise
    _raise(ex, cause)
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/_utils.py", line 413, in get_dataclass_data
    d[name] = _maybe_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/swelch/miniconda3/envs/isaacsim/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1087, in _node_wrap
    raise ValidationError(f"Unexpected type annotation: {type_str(ref_type)}")
omegaconf.errors.ValidationError: Unexpected type annotation: type[DigitalFilter]
    full_key: func
    object_type=DigitalFilterCfg
```

### System Info

Describe the characteristic of your environment:

- Commit: `92dcb8c05642aa890979a5e40a7b5cc9255799d2`
- Isaac Sim Version: `4.1.0-rc.7+4.1.14801.71533b68.gl`
- OS: `Ubuntu 20.04`
- GPU: `NVIDIA GeForce RTX 4060 Laptop GPU`
- CUDA: `12.2`
- GPU Driver: `535.183.01`

### Additional context

N/A

### Checklist

- [X] I have checked that there is no similar issue in the repo (**required**)
- [X] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

TBD

### PR DESCRIPTION
**Pull request by jtigue-bdai**
# Description

This PR add in the ability to properly convert configclass to dict if a configclass instance contains a list of configclasses.

Fixes #1219 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab/omni/isaac/lab/utils/dict.py [hunk 1]
```diff
@@ -55,6 +57,8 @@ def class_to_dict(obj: object) -> dict[str, Any]:
         # check if attribute is a dictionary
         elif hasattr(value, "__dict__") or isinstance(value, dict):
             data[key] = class_to_dict(value)
+        elif isinstance(value, list):
+            data[key] = [class_to_dict(v) for v in value]
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab/omni/isaac/lab/utils/dict.py]:
```suggestion
        elif isinstance(value, (list, tuple)):
            data[key] = type(value)([class_to_dict(v) for v in value])
```

File: source/extensions/omni.isaac.lab/test/utils/test_configclass.py
Comment (jtigue-bdai) [inline source/extensions/omni.isaac.lab/test/utils/test_configclass.py:1]: Test still WIP

**Comment by Mayankm96** [comment]
Thanks for the fix! Do we need to do something for dict to class operation too? Or that works as expected?

**Comment by jtigue-bdai** [comment]
Ah that is a good question. I will have to look at that.

**Comment by Dhoeller19** [comment]
@jtigue-bdai @Mayankm96 yes this works correctly and is tested here:
https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab/test/utils/test_configclass.py#L554

**Review by Dhoeller19 — APPROVED** [review]

**Pull request merged**